src/UrlHelper.php

```diff
@@ -102,6 +108,13 @@ public function __invoke(
             $routeParams = $this->mergeParams($routeName, $result, $routeParams);
         }
 
+        $reuseQueryParams = ! isset($options['reuse_query_params']) || (bool) $options['reuse_query_params'];
```

Comment:
Yikes! That's quite a BC break.

Am I reading this right? If I hit `/a?foo=bar` and then generate a URL for `b`, it'll end up `/b?foo=bar` by default? If so, that's not a behaviour I'd want 99.9999% of the time.

Reply:
Sorry should have mentioned that, yes it only applies to routes generated or current route, similar functionality to reuse_route_params. It will not reuse query params for different routes.